R/themeApa.R

```diff
@@ -0,0 +1,193 @@
+#'   left, bottom right--to place the legend. \code{ggplot2} provides no obvious
+#'   way to automatically choose a position that overlaps least with the geoms (the
+#'   plotted data), so users will need to choose one.
+#'
+#'   Facetting is supported, but APA guidelines are considerably less clear for
+#'   such situations.
+#'
+#'   This theme was created with inspiration from Rudolf Cardinal's
+#'   \href{http://egret.psychol.cam.ac.uk/statistics/R/graphs2.html}{code}, which
+#'   required updating for newer versions of \code{ggplot2} and adaptations for
+#'   APA style.
+#'
+#' @export
+themeApaRaw <- function(legend.pos       = getGraphOption("legend.position"),
+                        legend.use.title = FALSE,
+                        legend.font.size = getGraphOption("axis.title.cex"),
+                        font.size        = getGraphOption("fontsize"),
+                        x.font.size      = getGraphOption("fontsize"),
+                        y.font.size      = getGraphOption("fontsize"),
+                        facet.title.size = getGraphOption("fontsize"),
+                        remove.y.gridlines = TRUE,
+                        remove.x.gridlines = TRUE,
+                        family         = getGraphOption("family"),
+                        axisTickLength = getGraphOption("axisTickLength"),
+                        axisTickWidth  = getGraphOption("axisTickWidth")) {
+
+  theme <- ggplot2::theme_bw() +
+    ggplot2::theme(
+      plot.title           = ggplot2::element_text(family = family, face = "bold", hjust = 0, size = 14),
+      axis.title.x         = ggplot2::element_text(size = x.font.size),
+      axis.title.y         = ggplot2::element_text(size = y.font.size, angle = 90),
+      legend.text          = ggplot2::element_text(size = legend.font.size),
+      legend.key.size      = ggplot2::unit(1.5, "lines"),
+      legend.key           = ggplot2::element_blank(),
+      legend.key.width     = grid::unit(2,  "lines"),
+      strip.text.x         = ggplot2::element_text(size = facet.title.size),
+      strip.text.y         = ggplot2::element_text(size = facet.title.size),
+      strip.background     = ggplot2::element_rect(colour = "white", fill = "white"),
+      panel.background     = ggplot2::element_rect(fill = "white"),
+      plot.title.position  = "panel",
+
+      # additional options copied from themeJasp
+      text                 = element_text(family = family, size = font.size),
+      axis.ticks.length    = axisTickLength,
+      axis.ticks           = element_line(size = axisTickWidth, color = "black"))
+
+if (is.character(legend.pos)) {
+  theme <- theme + switch(legend.pos
+    "topleft"   = theme(legend.position = c(0.05, 0.95), legend.justification = c(0.05, 0.95)),
+    "topmiddle" = theme(legend.position = c(0.5, 0.95), legend.justification = c(0.5, 0.95)),
+    ...,
+    stop("unknown legend position string)
+  )
```

Comment:
For the switch you need to add all the cases of the if-else ladder. Also, there's a `"` missing after the `stop()`.

Reply:
oh, sorry. I thought that you meant the second PR, and assumed that there is some magical switch :D